rust/routee-compass-core/src/algorithm/search/a_star/a_star_algorithm.rs

```diff
@@ -188,6 +194,25 @@ pub fn run_a_star(
         solution.len()
     );
 
+    // check if DEBUG_MODE is set
+    if std::env::var("DEBUG_MODE").is_ok() {
+        let mut flamegraph = FlameGraphBuilder::default();
+        flamegraph.visit_root(&costs);
+        flamegraph.visit_root(&frontier);
+        flamegraph.visit_root(&traversal_costs);
+        flamegraph.visit_root(&solution);
+        let output = flamegraph.finish_and_write_flame_graph();
+
+        let search_name = match target {
+            None => format!("{}_to_all", source),
+            Some(tid) => format!("{}_to_{}", source, tid),
+        };
+
+        let mut flamegraph_file =
+            File::create(format!("search_memory_flamegraph_{}.out", search_name)).unwrap();
+        flamegraph_file.write_all(output.as_bytes()).unwrap();
```

Comment:
1. could we pass debug_mode as a parameter to the run_a_star method(s), and perform this env::var("_") check out in the SearchApp scope, so it only gets checked once
  2. could we set the filename based on the query source + target VertexIds so that if multiple queries are run, they all write flamegraphs to different destinations? or would that be silly?

Reply:
> could we pass debug_mode as a parameter to the run_a_star method(s), and perform this env::var("_") check out in the SearchApp scope, so it only gets checked once 

yeah that's smart; I also stumbled across [this SO answer](https://stackoverflow.com/a/39205417) about just doing conditional compilation for this section so we don't even include it in a release build.

> could we set the filename based on the query source + target VertexIds so that if multiple queries are run, they all write flamegraphs to different destinations? or would that be silly?

yeah I was wondering about this; I think it's setup now to write a file for each search by using the `search_name` variable which is a combo of the source and target ids.

Reply:
ya! debug_assertions! cool, let's do that!